ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -66,48 +67,140 @@ func readINI(dest *Destination) (*CLIConfig, map[interface{}]interface{}, error)
 	return iniReadWriter.GetConfig()
 }
 
-func readYAML(yamlPath string, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
-	// convert yaml to CliConfig
-	dat, err := ioutil.ReadFile(yamlPath)
+func readYAML(path string, configMap map[interface{}]interface{}) error {
+	// read cluster file
+	dat, err := ioutil.ReadFile(path)
 	if err != nil {
-		return errors.Wrapf(err, "Error reading yaml file %s", yamlPath)
-	}
-	if err = yaml.Unmarshal(dat, cliConfig); err != nil {
-		return errors.Wrapf(err, "Error parsing yaml file %s", yamlPath)
+		return err
 	}
 
-	// convert yaml to a map (replaces IsKeyPresent functionality)
+	// convert cluster yaml to a map (replaces IsKeyPresent functionality)
 	if err = yaml.Unmarshal(dat, &configMap); err != nil {
-		return errors.Wrapf(err, "Error parsing yaml file %s", yamlPath)
+		return err
 	}
-
 	return nil
 }
 
-// GetConfig gets the ecs-cli config object from the config file.
+// GetConfigs gets the ecs-cli config object from the config file(s).
 // map contains the keys that are present in the config file (maps string field name to string field value)
 // map is type map[interface{}]interface{} to ensure fowards compatibility with changes that will
 // cause certain keys to be mapped to maps of keys
-func (rdwr *YAMLReadWriter) GetConfig() (*CLIConfig, map[interface{}]interface{}, error) {
-	to := new(CLIConfig)
+// This function either reads the old single configuration file
+// Or if the new files are present, it reads from them instead
+func (rdwr *YAMLReadWriter) GetConfigs(clusterConfig string, profileConfig string) (*CLIConfig, map[interface{}]interface{}, error) {
+	cliConfig := &CLIConfig{}
 	configMap := make(map[interface{}]interface{})
 	// read the raw bytes of the config file
-	path := configPath(rdwr.destination)
+	profilePath := credentialsFilePath(rdwr.destination)
+	configPath := configFilePath(rdwr.destination)
+
+	// Try to read the config as YAML
+	clusterMap := make(map[interface{}]interface{})
```

Comment:
Yeah- unfortunately so far I have not come up with a solution that works using a struct.